plugins/modules/idrac_license.py

```diff
@@ -0,0 +1,590 @@
+from ansible.module_utils.urls import ConnectionError, SSLValidationError
+from ansible_collections.dellemc.openmanage.plugins.module_utils.utils import (
+    get_idrac_firmware_version, validate_and_get_first_resource_id_uri,
+    remove_key, idrac_redfish_job_tracking)
+
+SYSTEMS_URI = "/redfish/v1/Managers"
+IDRAC_JOB_URI = "{res_uri}/Jobs/{job_id}"
+EXPORT_LICENCE_NETWORK_SHARE_URI = "Oem/Dell/DellLicenseManagementService/Actions/DellLicenseManagementService.ExportLicenseToNetworkShare"
+LICENSE_URI = "/redfish/v1/LicenseService/Licenses/{license_id}"
+
+INVALID_LICENSE_MSG = "License id '{license_id}' is invalid."
+SUCCESS_EXPORT_MSG = "Successfully exported the license."
+SUCCESS_DELETE_MSG = "Successfully deleted the license."
+SUCCESS_IMPORT_MSG = "Successfully imported the license."
+FAILURE_MSG = "Unable to '{operation}' the license with id '{license_id}'."
+
+IGNORE_CERTIFICATE_WARNING = {"off": "Off", "on": "On"}
+PROXY_SUPPORT = {"off": "Off", "default_proxy": "DefaultProxy", "parameters_proxy": "ParametersProxy"}
+PROXY_TYPE = {"http": "HTTP", "socks": "SOCKS"}
+
+
+class DeleteLicense:
+    def __init__(self, idrac, module):
+        self.idrac = idrac
+        self.module = module
+
+    def execute(self, module):
+        license_id = module.params.get('license_id')
+        check_license_id(self, module, license_id, "delete")
+        delete_license_url = f"/redfish/v1/LicenseService/Licenses/{license_id}"
+        delete_license_response = self.idrac.invoke_request(delete_license_url, 'DELETE')
+        status = delete_license_response.status_code
+        if status == 204:
+            module.exit_json(msg=SUCCESS_DELETE_MSG, changed=True)
+        else:
+            module.exit_json(FAILURE_MSG.format(operation="delete", license_id=license_id), changed=False)
+        return delete_license_response
+
+
+class ExportLicense:
+    def __init__(self, idrac, module):
+        self.idrac = idrac
+        self.module = module
+
+    def execute(self, module):
+        share_type = module.params.get('share_parameters').get('share_type')
+        license_id = module.params.get('license_id')
+        check_license_id(self, module, license_id, "export")
+        if share_type == "local":
+            export_license_status = export_license_local(self, module)
+        elif share_type in ["http", "https"]:
+            export_license_status = export_license_http(self, module)
+            job_status = get_job_status(self, module, export_license_status)
+        elif share_type == "cifs":
+            export_license_status = export_license_cifs(self, module)
+            job_status = get_job_status(self, module, export_license_status)
+        elif share_type == "nfs":
+            export_license_status = export_license_nfs(self, module)
+            job_status = get_job_status(self, module, export_license_status)
+        status = export_license_status.status_code
+        if share_type in ["http", "https", "cifs", "nfs"]:
```

Comment:
Lets move ["http", "https", "cifs", "nfs"] to constant.
Infact This if statement to check local /network share can be removed if you initialize the job_status to empty in the beginning.

Reply:
Done